Modules/Core/Common/include/itkNeighborhoodInnerProduct.hxx

```diff
@@ -29,18 +29,15 @@ NeighborhoodInnerProduct<TImage, TOperator, TComputation>::Compute(const ConstNe
                                                                    const unsigned int                        start,
                                                                    const unsigned int                        stride)
 {
-  typename OperatorType::ConstIterator o_it;
-
   using InputPixelType = typename TImage::PixelType;
   using InputPixelRealType = typename NumericTraits<InputPixelType>::RealType;
   using AccumulateRealType = typename NumericTraits<InputPixelRealType>::AccumulateType;
 
-  AccumulateRealType sum = NumericTraits<AccumulateRealType>::ZeroValue();
-
   using OutputPixelValueType = typename NumericTraits<OutputPixelType>::ValueType;
 
-  o_it = op.Begin();
   const typename OperatorType::ConstIterator op_end = op.End();
+  typename OperatorType::ConstIterator       o_it = op.Begin();
+  AccumulateRealType                         sum = NumericTraits<AccumulateRealType>::ZeroValue();
```

Comment:
Would `AccumulateRealType sum{};` accomplish this?

Reply:
In general I believe that any `T var = NumericTraits<T>::ZeroValue()` can be replaced safely with `T var{}`.

Reply:
I'm working on a more universal patch to address this particular item.  I'd like to keep the universal change separate from the current patch set.

Note the old code used ZeroValue.